zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/migration/VersionCompatibilityCheckTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.state.migration;
+
+import static io.camunda.zeebe.engine.state.migration.VersionCompatibilityCheck.check;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.engine.state.migration.VersionCompatibilityCheck.CheckResult.Compatible;
+import io.camunda.zeebe.engine.state.migration.VersionCompatibilityCheck.CheckResult.Incompatible;
+import io.camunda.zeebe.engine.state.migration.VersionCompatibilityCheck.CheckResult.Indeterminate;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+@Execution(ExecutionMode.CONCURRENT)
+final class VersionCompatibilityCheckTest {
+
+  @Nested
+  final class IncompatibleResults {
+
+    @Test
+    public void shouldRejectDowngrades() {
+      assertThat(check("9.0.0", "8.0.0")).isInstanceOf(Incompatible.MajorDowngrade.class);
+      assertThat(check("9.0.0", "8.1.0")).isInstanceOf(Incompatible.MajorDowngrade.class);
+      assertThat(check("8.5.0", "8.4.0")).isInstanceOf(Incompatible.MinorDowngrade.class);
+      assertThat(check("8.5.0", "8.4.1")).isInstanceOf(Incompatible.MinorDowngrade.class);
+      assertThat(check("8.5.3", "8.5.2")).isInstanceOf(Incompatible.PatchDowngrade.class);
+      assertThat(check("8.5.3", "8.5.0")).isInstanceOf(Incompatible.PatchDowngrade.class);
+      assertThat(check("8.5.3", "8.5.3-alpha1"))
+          .isInstanceOf(Incompatible.PreReleaseDowngrade.class);
+      assertThat(check("8.1.0", "8.1.0-alpha1"))
+          .isInstanceOf(Incompatible.PreReleaseDowngrade.class);
```

Comment:
🔧 I'd love to a rejection for PreReleaseUpgrade as well (so from alpha to regular release

Reply:
These cases are covered as `ChangeOfPreReleaseVersion`, with matching tests.